canonical/charmed-kubeflow-uats
# feat: Add kfp-tensorflow notebook to confirm NVIDIA GPU access

Add `kfp-tensorflow` notebook in order to confirm access to a GPU. The notebook spins uses kfp SDK to create an experiment and a run that succeeds when:
 * the run's pod is scheduled on a node with an NVIDIA GPU
 * the run's code, and more specifically Tensorflow framework, has access to an NVIDIA GPU.

Closes #128

### Testing
Similar to #138, in order to test the PR
* Spin up a cluster with a GPU. For a two node microk8s node with one gpu, refer to this comment https://github.com/canonical/charmed-kubeflow-uats/issues/130#issuecomment-2465299876
* Run with the `--include-gpu-tests` flag e.g.
  ```
  tox -e uats-remote -- --include-gpu-tests --filter "kfp"
  ```
  This should run kfp cpu and gpu tests (kfp-v1, kfp-v2, kfp-tensorflow). You can also test from inside the notebook UI.
